### DISCUSSION TITLE
[@types/carbon__icons-react] outdated and incompatible with latest v11 version

### DISCUSSION THREAD

**Discussion by pReya** [DefinitelyTyped/DefinitelyTyped#61203] [Issues with a @types package] [answered]
The current types @types/carbon__icons-react are made for v10.XX.X. However the [project is on major version v11.5.0](https://github.com/carbon-design-system/carbon/tree/main/packages/icons-react) now. 

Unfortunately a lot has changed and the types aren't working anymore. Seems they have removed the "per-size"-components like `<Add32>` and now it's all `<Add size=32>`etc.

[a comment by an automated account is not shown]

**Comment by pReya** [comment]
Hey @metonym thanks for your initial contribution! Any chance you might find time to update the types to v11? I'd love to help, but I've never created a DT package and I am not sure where/how to start?

**Reply by metonym** [reply]
It looks like Carbon will ship their own Type definitions for `@carbon/icons-react` v11: https://github.com/carbon-design-system/carbon/pull/11533

However, it may still be useful to update this package to reflect the v11 API.

**Comment by metonym** [comment, marked as answer]
@pReya The types have been updated to [v11.5.0](https://www.npmjs.com/package/@types/carbon__icons-react).

<img width="600" alt="@types:carbon__icons-react@11 5 0" src="https://user-images.githubusercontent.com/10718366/182986702-6fda0202-d640-49d7-8cb4-10906eb227ea.png">

**Reply by pReya** [reply]
Unfortunately, even with the latest version 11.6 I'm still getting TypeScript errors on the import. What am I missing?
<img width="1585" alt="Screenshot 2022-08-05 at 10 28 09" src="https://user-images.githubusercontent.com/4677417/183036414-254ae70d-20fc-4014-9cee-c72f69e81fe5.png">

**Reply by metonym** [reply]
I'm unfamiliar with your set-up, but here are a couple things to try:

- Do you have `@types/carbon__icons-react@11.5.0` installed?
- Have you tried restarting the TS language server?

**Reply by pReya** [reply]
Nevermind, I misunderstood your original comment and thought, the types were now included in the original package. I've now installed the types package and everything works perfect now. Thank you so much!

**Reply by metonym** [reply]
FWIW, there is a [closed PR](https://github.com/carbon-design-system/carbon/pull/11533#issuecomment-1205968217) in the upstream Carbon repo that would have added the types to the package itself.

It's an ongoing effort to see if and how the source library can support TypeScript. Once supported upstream, we can finally remove the `@types/carbon__icons-react` package. In the meanwhile, we'll need to periodically update the types.

**Comment by metonym** [comment]
As of version 11.29.0, `@carbon/icons-react` ships with its own TypeScript definitions.

As such, `@types/carbon__icons-react` is no longer needed and has [been removed](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67176) as of v11.29.0.
